File: src/commands/settingsCommands.ts
// src/commands/settingsCommands.ts 
import { core } from "@tauri-apps/api";
import type { Settings } from "../types";

export const settingsCommands = {
    async getSettings(): Promise<Settings> {
        return await core.invoke("get_settings");
    },

    async updateSettings(newSettings: Settings): Promise<void> {
        return await core.invoke("update_settings", {
            newSettings
        });
    },
};

File: src-tauri/src/commands/settings_commands.rs
// src/commands/settings_commands.rs

use crate::models::Settings;
use crate::states::AppState;
use std::sync::Arc;
use tauri::State;

#[tauri::command]
pub async fn get_settings(app_state: State<'_, Arc<AppState>>) -> Result<Settings, String> {
    app_state.get_settings().map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn update_settings(
    new_settings: Settings,
    app_state: State<'_, Arc<AppState>>,
) -> Result<(), String> {
    app_state
        .update_settings(new_settings)
        .await
        .map_err(|e| e.to_string())
}
